### ISSUE TITLE
Linux compilation errors related to smithay-client-toolkit

### ISSUE THREAD

**Issue by AustinJ235** [rust-windowing/winit#1562]
**OS: Arch Linux
GPU: Amd Vega 56 & Nvidia GTX 780**

The version that winit uses of the lib smithay-client-toolkit seems to be a bit dated. This dependancy brings in wayland-protocols which fails to build on my linux systems.

```
   --> /home/austin/Workspace/vk-voxel/target/release/build/wayland-protocols-aba19884de702452/out/fullscreen-shell-v1_client_api.rs:728:1
    |
655 |         fn capability(&mut self, object: ZwpFullscreenShellV1, capability: u32) {}
    |                                                                                 -- this block is empty, you might have not meant to close it
...
728 | }
    | ^ unexpected closing delimiter

   --> /home/austin/Workspace/vk-voxel/target/release/build/wayland-protocols-aba19884de702452/out/keyboard-shortcuts-inhibit-v1_client_api.rs:362:1
    |
313 |     pub trait EventHandler {}
    |                            -- this block is empty, you might have not meant to close it
...
362 | }
    | ^ unexpected closing delimiter

   --> /home/austin/Workspace/vk-voxel/target/release/build/wayland-protocols-aba19884de702452/out/linux-explicit-synchronization-v1_client_api.rs:368:1
    |
322 |     pub trait EventHandler {}
    |                            -- this block is empty, you might have not meant to close it
...
368 | }
    | ^ unexpected closing delimiter
```

I can manually fix these errors as they seem to be code generation problems. There are functions that begin within comments that shouldn't be in the comments.

It has been fairly annoying to manually go through and change these files everytime I do a fresh build on my linux systems. I am not sure if this would be solved by updating the lib used. These does seem to be a fairly large task just taking a glance at it, but I think this is something that should be looked at.

**Comment by kchibisov** [comment]
Could you explain how are you compiling winit, are you passing some special flags to do certain stuff? winit compiles just fine and tested on Ci on Linux, so something like that would be noticed long time ago. I'm not even sure how to repro it.

cc @vberger

**Comment by AustinJ235** [comment]
Interesting so if I compile it alone it works fine, if I compile `vulkano-win` it works fine, but If I compile [basalt](https://github.com/AustinJ235/basalt) I get the errors. I don't use any additional features. Maybe if someone else could give one of my examples a go in my repo that'd be appreciated. All of the winit code used in my lib is within one file, ``basalt/src/window/winit.rs``

**Comment by kchibisov** [comment]
I can compile your lib, but I had to delete `rustfmt.toml`, I guess. It seems like someone is trying to run formatter on bindings, which is failing due to fmt bugs (IIRC I've seen that issue in the past already). But default it shouldn't run formatter, I guess.

**Comment by vberger** [comment]
You are hitting this issue: https://github.com/Smithay/wayland-rs/issues/291

TL;DR : wayland-scanner runs rustfmt on the generated code if available, to ease debugging (otherwise the whole thing is generated on a single line which is horrible to inspect).

However rustfmt's `normalize_doc_attributes = true` is bugged, and comments out parts of the generated code during the formatting: https://github.com/rust-lang/rustfmt/issues/3988
